src/dmd/dtemplate.d

```diff
@@ -8173,3 +8173,61 @@ MATCH matchArg(TemplateParameter tp, Scope* sc, RootObject oarg, size_t i, Templ
     else
         assert(0);
 }
+
+
+/***********************************************
+ * Collect and print statistics on template instantiations.
+ */
+struct TemplateStats
+{
+    __gshared TemplateStats[void*] stats;
+
+    uint numInstantiations;     // number of instantiations of the template
```

Comment:
Are we sure this'll never overflow? Should we be using checkedint addu intrinsics?

Reply:
It'll overflow at 4 billion instantiations. Your compile will likely fail loooong before that wraps. Even if it does wrap, it won't hurt anything.

Reply:
I seem to recall of one of the commercial projects instantiating 200GB worth of templates, so I don't think 4 billion is too big a number to assume failure. :-)

Reply:
yeah but it doesnt cost much to add a compiler error on overflow

Reply:
Yeah most PCs use 64-bit anyhow, so I don't really see why this couldn't at least be `size_t`.

Reply:
Where are you going to find the memory to store the instantiations of 4 billion templates?

Reply:
There are machines with more than a Terabyte of RAM these days.

And yes for template heavy code you need those ;-)

Reply:
Do you really imagine the storage costs per instantiation is less than 256 bytes?

Might as well argue the line counter should be a ulong. Or the column number.

Seriously, folks, these are not going to wrap, and even if they do, it *doesn't matter*.